pkg/api/query_range.go

```diff
@@ -89,6 +89,7 @@ func queryRange(promqlConf *query.Config, queryEngine *promql.Engine, queryable
 
 		qry, err := queryEngine.NewRangeQuery(
 			queryable,
+			&promql.QueryOpts{EnablePerStepStats: true},
```

Comment:
Can you explain why we want stats enabled? Do we use these stats?

Reply:
This allows the stats feature to work.

https://github.com/timescale/promscale/blob/343bfa83afcc05f263ef1a39b4d5cc6db4593325/pkg/promql/engine.go#L479

If you see how the feature works, there is a logical AND condition between
1. Input of `-enable-feature=promql-per-step-stats` that is supplied in the CLI (`ng.enablePerStepStats` part)
2. Input from the caller creating query (range & instant) (`opts.EnablePerStepStats` part). This one is the line you commented on

So, the feature will work only when `1 && 2 == true`. `1` is based on user, `2` is Promscale enabling the feature. Hence, we pass in `&promql.QueryOpts{EnablePerStepStats: true}` so that the stats feature is now enabled.

> why we want stats enabled?

It's a PromQL feature that we were lacking. Prometheus provides it, but we don't, as it was recently merged into Prometheus master. With this PR, we are again up to date with the master.

Reply:
Note: We do not enable this, the enabling will happen only when the user passes in `-enable-feature=promql-per-step-stats`